packages/consensus/src/validation/script.cairo

```diff
@@ -59,21 +66,44 @@ impl EngineTransactionDummyImpl of EngineTransactionTrait<Transaction, TxIn, TxO
 
 fn script_flags(header: @Header, tx: @Transaction) -> u32 {
     let mut script_flags = 0_u32;
+    let block_height = tx.inputs[0].previous_output.block_height;
 
-    if true { // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1208
+    // Blocks created after the BIP0016 activation time need to have the
+    // pay-to-script-hash checks enabled.
+    if block_height >= @BIP_16_BLOCK_HEIGHT {
         script_flags += ScriptFlags::ScriptBip16.into();
     }
 
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1215
-    
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1221
+    // Enforce DER signatures for block versions 3+
+    // This is part of BIP0066.
+    if header.version >= @3_u32 && block_height >= @BIP_66_BLOCK_HEIGHT {
+        script_flags += ScriptFlags::ScriptVerifyDERSignatures.into();
+    }
 
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1234
+    // Enforce CHECKLOCKTIMEVERIFY for block versions 4+
+    // This is part of BIP0065.
+    if header.version >= @4_u32 && block_height >= @BIP_65_BLOCK_HEIGHT {
+        script_flags += ScriptFlags::ScriptVerifyCheckLockTimeVerify.into();
+    }
 
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1266
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1267
+    // Enforce CHECKSEQUENCEVERIFY if the CSV soft-fork is now active
+    //  This is part of BIP0112.
+    if block_height >= @BIP_112_BLOCK_HEIGHT {
+        script_flags += ScriptFlags::ScriptVerifyCheckSequenceVerify.into();
+    }
 
-    // TODO: https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/blockchain/validate.go#L1279
+    // Enforce the segwit soft-fork
+    // This is part of BIP0141.
+    if block_height >= @BIP_141_BLOCK_HEIGHT {
+        script_flags += ScriptFlags::ScriptVerifyWitness.into();
```

Comment:
It seems: https://github.com/bitcoin/bitcoin/blob/6a370435526875a441001be8c44c9b94a2361c8c/src/validation.cpp#L2393 like one more flag is necessary.

Reply:
the `SCRIPT_VERIFY_NULLDUMMY` flag seem to be part of the `ScriptStrictMultiSig` flag in shinigami ScriptFlags: https://github.com/keep-starknet-strange/shinigami/blob/13f6cf41c6ea1d3c229a2f10ce2ddb0a4626c6cc/packages/engine/src/scriptflags.cairo#L93